src/main/java/dukehandler/TaskManager.java

```diff
@@ -0,0 +1,212 @@
+    static ArrayList<Task> tasks = new ArrayList<>();
+
+    public TaskManager() {
+    }
+    public static void loadTasksFromFile(String filePath) throws FileNotFoundException {
+        File f = new File(filePath); // create a File for the given file path
+        Scanner s = new Scanner(f); // create a Scanner using the File as the source
+        while (s.hasNext()) {
+            String[] part = s.nextLine().split(" ~~ ");
+            switch(part[0]) {
+            case "T":
+                tasks.add(new ToDo(part[2]));
+                break;
+            case "D":
+                tasks.add(new Deadline(part[2], part[3]));
+                break;
+            case "E":
+                tasks.add(new Event(part[2], part[3]));
+                break;
+            }
+            if (part[1].equals("X")) {
+                tasks.get(tasks.size()-1).markAsDone();
+            }
+        }
+    }
+
+    public static void saveTasksToFile(String filePath) throws IOException {
+        FileWriter fw = new FileWriter(filePath);
+        for (Task task : tasks) {
+            fw.write(task.getTaskType() + " ~~ "
+                    + task.getStatusIcon() + " ~~ "
+                    + task.getTaskName());
+            if (task.getTaskType().equals("D") || task.getTaskType().equals("E")) {
+                fw.write(" ~~ " + task.getTime());
+            }
+            fw.write(System.lineSeparator());
+        }
+        fw.close();
+    }
+
+    public static void printAllTasks() {
+        boolean isAllDone = true;
+        if (tasks.isEmpty()) {
+            MessagePrinter.printEmptyListMessage();
+            return;
+        }
+        System.out.println(" Here are the tasks in your list:");
+        for (int i = 1; i < tasks.size() + 1; ++i) {
+            if ((tasks.get(i - 1).getStatusIcon()).equals(" ")) {
+                isAllDone = false;
+            }
+            System.out.println(" " + i + "." + tasks.get(i - 1).toString());
+        }
+        if (isAllDone) {
+            System.out.println(" Woah, all completed! Good job!");
+        }
+    }
+
+    public static void markTaskAsDone(String doneIndexString) {
+        int doneIndexInteger;
+        try {
+            doneIndexInteger = Integer.parseInt(doneIndexString);
+            checkTaskToMarkDone(doneIndexInteger);
+        } catch (ExceptionEmptyList eel) {
+            MessagePrinter.printEmptyListMessage();
+            return;
+        } catch (ExceptionIllegalTaskMarkedDone | NumberFormatException nfe) {
+            MessagePrinter.printDoneOrDeleteTaskErrorMessage("done", tasks.size());
+            return;
+        } catch (ExceptionAlreadyTaskMarkedDone atmd) {
+            System.out.println(" Hey, you've already marked that as done!");
+            return;
+        }
+
+        tasks.get(doneIndexInteger - 1).markAsDone();
+        System.out.println(" Nice! I've marked this task as done:\n "
+                + tasks.get(doneIndexInteger - 1).toString());
```

Comment:
Perhaps leave all message printing to the MessagePrinter class? So you can refactor this into a function there and maintain SLAP.